### ISSUE TITLE
Support Azure AI Model Inference API provider 

### ISSUE THREAD

**Issue by wong2** [vercel/ai#2395]
### Feature Description

Today I deployed the Llama 3.1 model using Azure AI Studio and got it ready to use through the AI SDK. I originally thought its API was compatible with OpenAI, but it didn't work out when I tried. 
From the [documentation](https://learn.microsoft.com/en-us/azure/ai-studio/reference/reference-model-inference-api?tabs=javascript), I found out it uses the Azure AI Model Inference API, so does that mean we need a separate provider?

**Comment by lgrammel** [comment]
@wong2 yes this might require a new provider, like `azure-inference`

**Comment by wong2** [comment]
This need has come up again because we want to use DeepSeek R1 on Azure.

**Comment by DakotaWray2** [comment]
Agreed we would really benefit from a non-OpenAI azure provider so we can use Azure DeepSeek!

**Comment by DrPye** [comment]
+1 for an inference api provider!

**Comment by alexcoimbra12** [comment]
I made the following code and it works for me.

```js
import { createOpenAICompatible } from '@ai-sdk/openai-compatible';

const API_VERSION = process.env.AZURE_API_VERSION
const AZURE_API_KEY = process.env.AZURE_API_KEY
const AZURE_RESOURCE_NAME = process.env.AZURE_RESOURCE_NAME

const azureInf = createOpenAICompatible({
  name: 'azure-inf',
  baseURL: `https://${AZURE_RESOURCE_NAME}.services.ai.azure.com/models`,
  queryParams: {
    'api-version': API_VERSION,
  },
  fetch: async (url, request) => {
    return await fetch(url, { ...request, headers: {...request?.headers, 'api-key': AZURE_API_KEY} });
  },
});

.......

// Then you can use here
'chat-model-reasoning': wrapLanguageModel({
      model: azureInf('DeepSeek-R1'),
      middleware: extractReasoningMiddleware({ tagName: 'think' }),
    }),
```

**Comment by heiwen** [comment]
Giving this a bump.

My understanding is that Foundry is Microsoft's preferred way to access AI models (including OpenAI models), similar to Bedrock and Vertex.

The Azure OpenAI Service on the other hand is a more specialized service which could be compared with the Anthropic sub-provider in Bedrock / Vertex.

This implies that the naming of the current provider is actually slightly misleading / inconsistent with the Vertex / Bedrock ones.

Some more info: https://learn.microsoft.com/en-us/answers/questions/5572779/azure-openai-or-azure-ai-foundry

**Comment by khalidsaidi** [comment]
A2ABench has an accepted answer for this imported thread.
- Thread: https://a2abench-api.web.app/q/cmn1ngy830073xzty0i3enwyp
- Accepted at: 2026-03-25T19:53:02.459Z
- Accepted answer agent: `ai village`
- Answer preview: "No matching threads were found for this query."

**Comment by turisanapo** [comment]
Update: Azure has deprecated the AI Inference SDK and [will retire it on May 30, 2026](https://learn.microsoft.com/en-us/azure/ai-foundry/foundry-models/supported-languages?view=foundry-classic). Microsoft recommends switching to the OpenAI-compatible endpoint via the [OpenAI/v1 API](https://aka.ms/openai/v1).

This means `createAzure` from `@ai-sdk/azure` should work for all Azure-hosted models — including non-OpenAI ones like Claude, Llama, and Mistral. A dedicated `@ai-sdk/azure-inference` provider may no longer be needed.

**Comment by lgrammel** [comment]
### Identify issue type

**Identified type: Feature**  
Confidence: High

Rationale: The issue primarily requests new provider/model support for Azure AI Model Inference/Azure AI Studio rather than reporting a broken documented AI SDK behavior.

Evidence:
- Title asks to support an Azure AI Model Inference API provider.
- Description says Azure AI Studio Llama 3.1 did not work with the OpenAI-compatible assumption and asks whether a separate provider is needed.
- Maintainer comment says this might require a new provider like `azure-ai-studio`.
- Multiple comments request support for Azure-hosted DeepSeek and an inference API provider.
- Later comment notes Azure may now recommend OpenAI-compatible endpoints and a dedicated provider may no longer be needed.

**Comment by lgrammel** [comment]
## Feature Availability

**Feature is available in AI SDK 7.0.14**

The requested support for non-OpenAI Azure AI Foundry deployments is available. `createAzure` accepts a Foundry OpenAI/v1 base URL without rewriting it, `azure.chat()` supports compatible deployments such as Llama, and `azure.deepseek()` provides Azure-hosted DeepSeek reasoning and streaming support. The repository includes generateText/streamText DeepSeek examples, legacy native AI-inference provider documentation, and 62 passing Azure provider tests covering custom base URLs and DeepSeek behavior.

### Documentation

- [Azure OpenAI](https://ai-sdk.dev/providers/ai-sdk-providers/azure)
- [Azure AI](https://ai-sdk.dev/providers/community-providers/azure-ai)
- [Get started with DeepSeek R1](https://ai-sdk.dev/resources/recipes/guides/r1)

### Related PRs

- [#4909: feat (docs): add azure custom provider](https://github.com/vercel/ai/pull/4909)
- [#15942: feat(azure): add deepseek model support to azure](https://github.com/vercel/ai/pull/15942)
- [#16649: fix(azure): use custom base urls if provided](https://github.com/vercel/ai/pull/16649)
